### PR TITLE
add plot_profile function

### PR DESCRIPTION
**Pull request by mafrahm**
This task adds the possibilty to create profile plots

```
        law run cf.PlotVariables1D --version prod1 --processes st_tchannel_t \
            --variables jet1_pt-jet2_pt \
            --plot-function columnflow.plotting.plot_functions_1d.plot_profile
```
(tested with hh2bbww analysis)
plot__proc_st_tchannel_t__cat_incl__var_jet1_pt-jet2_pt

I also fixed a minor inconsistency with plot colors when no color was set for a process

### CODE REVIEW

**Comment by mafrahm** [comment]
I also added the possibility to add the base distribution of the profiled variable. Here is another exemplary plot

profile1

File: columnflow/plotting/plot_functions_1d.py [hunk 1]
```diff
@@ -298,3 +300,142 @@ def plot_cutflow(
...
+) -> plt.Figure:
+    """
+    Takes 2-dimensional histograms as an input and profiles over the second axis.
+
+    This task adds two custom parameters, *skip_base_distribution* and *base_distribution_yscale*,
+    that can be selected on command-line via the --general-settings parameter
+
+    Exemplary task call:
+
+    .. code-block:: bash
```
Comment (riga) [inline columnflow/plotting/plot_functions_1d.py:329]: `code-block` needs to be followed by an empty line for sphinx to render this properly.

File: columnflow/plotting/plot_util.py [hunk 2]
```diff
@@ -392,3 +392,96 @@ def reduce_with(spec: str | float | callable, values: list[float]) -> float:
...
+
+    with np.errstate(invalid="ignore"):
+        mean = num / den
+        _mean = broadcast_nminus1d_to_nd(mean, values.shape, axis)
+        width = np.sum(values * (centers - _mean) ** 2, axis=axis) / den
+
+    return mean, width
+
+
+def get_profile_variations(h_in: hist.Hist, axis: int = 1) -> dict(str, hist.Hist):
```
Comment (riga) [inline columnflow/plotting/plot_util.py:461]: The return type looks fishy. I guess `dict[str, hist.Hist]`?

File: columnflow/plotting/plot_util.py [hunk 3]
```diff
@@ -392,3 +392,96 @@ def reduce_with(spec: str | float | callable, values: list[float]) -> float:
...
+    initial_shape = list(x.shape)
+    initial_shape.insert(axis, 1)
+
+    x = np.reshape(x, initial_shape)
+    x = np.broadcast_to(x, final_shape)
+
+    return x
+
+
+def get_profile_width(h_in: hist.Hist, axis: int = 1) -> (np.array, np.array):
```
Comment (riga) [inline columnflow/plotting/plot_util.py:439]: Return type should be `tuple[np.array, np.array]`.

**Review by riga — CHANGES_REQUESTED** [review]
Looks good!
Just 3 minor things inline.

**Review by riga — APPROVED** [review]

**Pull request merged**